### DISCUSSION TITLE
Applying  compute_matrix_by_fitting_to_plane filter on a meshset results in PyMeshLabException

### DISCUSSION THREAD

**Discussion by sizekick-manu** [cnr-isti-vclab/PyMeshLab#260] [Q&A]
I am trying to apply a filter compute_matrix_by_fitting_to_plane on a MeshSet object as follows:

ms.compute_matrix_by_fitting_to_plane(targetplane=int(0), rotaxis=int(0), toorigin=True, freeze=True, alllayers=False).

But I keep getting the error :
PyMeshLabException: Failed to apply filter: compute_matrix_by_fitting_to_plane
Details: Cannot compute rotation: there is no selection

Why do I get this error ? I am able to apply other filters such as compute_matrix_by_fitting_to_plane, compute_matrix_by_principal_axis etc. 

My main objective is to align a body mesh to the X-axis.

**Comment by antarikshnarain** [comment]
You need to select the mesh
```
ms.apply_filter(filter_name="set_selection_all",allfaces=True)
ms.apply_filter(filter_name="compute_matrix_by_fitting_to_plane", targetplane=0, rotaxis=0, toorigin=True, freeze=True)
```
